src/ipAllowed.js

```diff
@@ -11,22 +11,45 @@ module.exports = function(allowedSubnets) {
   assert(allowedSubnets, 'Must provide allowed subnets in CIDR format');
   assert(Array.isArray(allowedSubnets), 'Allowed subnets must be a list');
 
-  let subnets = [];
+  let subnetsV4 = [];
+  let subnetsV6 = [];
 
   for (let subnet of allowedSubnets) {
-    subnets.push(ipaddr.parseCIDR(subnet));
+    let parsed = ipaddr.parseCIDR(subnet);
+    // Let's be sure that this value is a 2-tuple of an object and a mask
+    // length.  Probably overkill
+    assert(typeof parsed[0] === 'object');
+    assert(typeof parsed[1] === 'number');
+    assert(parsed.length === 2);
+    if (parsed[0].kind() === 'ipv4') {
+      subnetsV4.push(parsed);
+    } else if (parsed[0].kind() === 'ipv6') {
+      subnetsV6.push(parsed);
+    } else {
+      assert(false, 'CIDR Specifies unknown address type');
+    }
   }
 
   return function(ip) {
     // Parse the IP into an IP object
     ip = ipaddr.parse(ip);
 
-    // Go through the subnets and see if we can match any.  A match means the
-    // ip is in the subnet
-    for (let subnet of subnets) {
-      if (ip.match(subnet)) {
-        return true;
+    // We want to match IPv4 addresses to allowed IPv4 subnets and IPv6 to
+    // allowed IPv6 subnets
+    if (ip.kind() === 'ipv4') {
+      for (let subnet of subnetsV4) {
+        if (ip.match(subnet)) {
+          return true;
+        }
+      }
+    } else if (ip.kind() === 'ipv6') {
+      for (let subnet of subnetsV6) {
+        if (ip.match(subnet)) {
+          return true;
+        }
       }
+    } else {
+      throw new Error('Ip address is unexpected format: ' + ip.kind());
```

Comment:
this code path should be extremely rare, so the extra call to `.kind()` shouldn't be a concern.  For us to hit that code path, `ipaddr.js` would have to add a new type of IP address.  This is mainly to catch us upgrading `ipaddr.js` but not accounting for the changes